Lib/fontTools/feaLib/builder.py

```diff
@@ -1369,14 +1366,27 @@ def add_single_subst_chained_(self, location, prefix, suffix, mapping):
         # https://github.com/fonttools/fonttools/issues/512
         # https://github.com/fonttools/fonttools/issues/2150
         chain = self.get_lookup_(location, ChainContextSubstBuilder)
-        sub = chain.find_chainable_single_subst(mapping)
+        sub = chain.find_chainable_subst(mapping, SingleSubstBuilder)
         if sub is None:
             sub = self.get_chained_lookup_(location, SingleSubstBuilder)
         sub.mapping.update(mapping)
         chain.rules.append(
             ChainContextualRule(prefix, [list(mapping.keys())], suffix, [sub])
         )
 
+    def add_multi_subst_chained_(self, location, prefix, glyph, suffix, replacements):
+        if not all(prefix) or not all(suffix):
+            raise FeatureLibError(
+                "Empty glyph class in contextual substitution", location
```

Comment:
what are prefix and suffix at this point? lists of glyph classes?

Reply:
yes, list of sets (of glyph names). I simply copied the logic from `add_single_subst_chained_` immediately above this